javers-persistence-sql/src/main/java/org/javers/repository/sql/domain/GlobalIdRepository.java

```diff
@@ -17,49 +19,58 @@
     private JsonConverter jsonConverter;
 
     public GlobalIdRepository(JaversPolyJDBC javersPolyjdbc, JsonConverter jsonConverter) {
-
         this.javersPolyjdbc = javersPolyjdbc;
         this.jsonConverter = jsonConverter;
     }
 
-    public Object save(GlobalId globalId) {
+    public Long save(GlobalId globalId) {
+
+        Long lookup = getIfExists(globalId);
+
+        if (lookup != null) {
+            return lookup;
+        }
+
+        return insert(globalId);
+    }
 
-        SelectQuery selectQuery = javersPolyjdbc.query().select().query("SELECT COUNT (" + FixedSchemaFactory.GLOBAL_ID_PK + ") \n" +
-                "FROM \n" +
-                "  public.jv_global_id, \n" +
-                "  public.jv_cdo_class\n" +
-                "WHERE \n" +
-                "  jv_global_id.cdo_class_fk = jv_cdo_class.cdo_class_pk AND\n" +
-                "  jv_global_id.local_id = '" + jsonConverter.toJson(globalId.getCdoId()) + "'");
+    private Long getIfExists(GlobalId globalId) {
 
+        //todo: z tym coś trzeba wymyślić!
+        SelectQuery selectQuery = javersPolyjdbc.query()
+                .select(GLOBAL_ID_PK)
+                .from(GLOBAL_ID_TABLE_NAME + "," + CDO_CLASS_TABLE_NAME)
+                .where(GLOBAL_ID_TABLE_NAME + "." + GLOBAL_ID_CLASS_FK + " = " + CDO_CLASS_TABLE_NAME + "." + CDO_CLASS_PK
+                        + " AND " + GLOBAL_ID_TABLE_NAME + "." + GLOBAL_ID_LOCAL_ID + " = '" + jsonConverter.toJson(globalId.getCdoId()) + "'");
 
-        javersPolyjdbc.queryRunner().queryList(selectQuery, new ObjectMapper<Integer>() {
+        List<Long> global_id_pks = javersPolyjdbc.queryRunner().queryList(selectQuery, new ObjectMapper<Long>() {
             @Override
-            public Integer createObject(ResultSet resultSet) throws SQLException {
-                return resultSet.getInt("global_id_pk");
+            public Long createObject(ResultSet resultSet) throws SQLException {
+                return resultSet.getLong("global_id_pk");
             }
         });
 
         if (javersPolyjdbc.queryRunner().queryExistence(selectQuery)) {
```

Comment:
hmm jak wcześniej pobieramy ids to już chyba będziemy wiedzieć po tym czy jest id, bez drugiego zapytania